waku/waku_filter_v2/client.nim

```diff
@@ -26,44 +26,56 @@ proc sendSubscribeRequest(
+  except CatchableError:
+    let errMsg = "failed to dialPeer: " & getCurrentExceptionMsg()
+    trace "failed to dialPeer", error = getCurrentExceptionMsg()
+    waku_filter_errors.inc(labelValues = [errMsg])
+    return err(FilterSubscribeError.badResponse(errMsg))
 
   let connection = connOpt.get()
 
-  # TODO: this can raise an exception
-  await connection.writeLP(filterSubscribeRequest.encode().buffer)
-
-  let respBuf = await connection.readLp(DefaultMaxSubscribeResponseSize)
-  let respDecodeRes = FilterSubscribeResponse.decode(respBuf)
-  if respDecodeRes.isErr():
-    trace "Failed to decode filter subscribe response", servicePeer
-    waku_filter_errors.inc(labelValues = [decodeRpcFailure])
-    return err(FilterSubscribeError.badResponse(decodeRpcFailure))
-
-  let response = respDecodeRes.get()
-
-  if response.requestId != filterSubscribeRequest.requestId:
-    trace "Filter subscribe response requestId mismatch", servicePeer, response
-    waku_filter_errors.inc(labelValues = [requestIdMismatch])
-    return err(FilterSubscribeError.badResponse(requestIdMismatch))
-
-  if response.statusCode != 200:
-    trace "Filter subscribe error response", servicePeer, response
-    waku_filter_errors.inc(labelValues = [errorResponse])
-    let cause =
-      if response.statusDesc.isSome():
-        response.statusDesc.get()
-      else:
-        "filter subscribe error"
-    return err(FilterSubscribeError.parse(response.statusCode, cause = cause))
+  try:
+    await connection.writeLP(filterSubscribeRequest.encode().buffer)
+
+    let respBuf = await connection.readLp(DefaultMaxSubscribeResponseSize)
+    let respDecodeRes = FilterSubscribeResponse.decode(respBuf)
+    if respDecodeRes.isErr():
+      trace "Failed to decode filter subscribe response", servicePeer
+      waku_filter_errors.inc(labelValues = [decodeRpcFailure])
+      return err(FilterSubscribeError.badResponse(decodeRpcFailure))
+
+    let response = respDecodeRes.get()
+
+    if response.requestId != filterSubscribeRequest.requestId:
+      trace "Filter subscribe response requestId mismatch", servicePeer, response
+      waku_filter_errors.inc(labelValues = [requestIdMismatch])
+      return err(FilterSubscribeError.badResponse(requestIdMismatch))
+
+    if response.statusCode != 200:
+      trace "Filter subscribe error response", servicePeer, response
+      waku_filter_errors.inc(labelValues = [errorResponse])
+      let cause =
+        if response.statusDesc.isSome():
+          response.statusDesc.get()
+        else:
+          "filter subscribe error"
+      return err(FilterSubscribeError.parse(response.statusCode, cause = cause))
+  except CatchableError:
+    let errMsg = "exception in waku_filter_v2 client: " & getCurrentExceptionMsg()
+    trace "exception in waku_filter_v2 client", error = getCurrentExceptionMsg()
+    waku_filter_errors.inc(labelValues = [errMsg])
+    return err(FilterSubscribeError.badResponse(errMsg))
```

Comment:
wouldn't it be better for the `try` block to be more scoped?

Reply:
> wouldn't it be better for the `try` block to be more scoped?

Yes, you are absolutely right! I tried to apply that in https://github.com/waku-org/nwaku/pull/2696/commits/06546bbd6cd782139adb3a0944e3b55e34da4996